extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -49,10 +65,23 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
 			}
 			this.formContainer.addItems(components);
 		} else {
+			// Options of a boolean dropdown
+			this.booleanOptionsArray = Object.keys(BooleanDropdownOptions);
 			// Initilaize general Tab sections
 			this.initializeBackupSection();
 			this.initializeDatabaseSection();
 
+			//Initilaize options Tab sections
+			this.initializeOptionsGeneralSection();
+			this.initializeAutomaticSection();
+			if (this.objectInfo.isLedgerDatabase !== null) {
```

Comment:
We generally should just be using undefined, not null. And typically for these checks just doing a strict check (e.g. for this `isLedgerDatabase === true`) would be preferred to avoid potential issues with mixed null/undefined usage

```suggestion
			if (this.objectInfo.isLedgerDatabase === true) {
```

Reply:
This condition is for to check if the property is supported or unsupported. In 2019 server we do not show this property in UI at all, in 2022 it could be either true/false. doing a strict check would not help to hide the option for unsupported servers right? or am i missing something? thanks

Reply:
Ok, if this is specifically just to check whether to show the value at all then you should use the isUndefinedOrNull function I linked below